preprocessing.py

```diff
@@ -76,23 +76,7 @@
             metadata["description"] = None
             metadata["schema"] = str(schema[i])
             featureset_metadata.append(metadata)
-        
-        # Convert featureset metadata (featurespec) to yaml
-        featureset_metadata = yaml.dump(featureset_metadata, default_flow_style=False)
-        with open("fspec.yaml", "w") as f:
-            f.write(featureset_metadata)
-        # Upload featureset metadata (featurespec)
-        resp = api.upload_featurespec(featureset=fs[k], filepath="fspec.yaml")
-        print("featurespec upload response:", resp)
-        
-        ########--- Commit features ---########
-        # Featureset
-        featureset = DkubeFeatureSet()
-        # Specify features path - mounted as output
-        featureset.update_features_path(path=out_path[k])
-        # Write features - Dataframe
-        featureset.write(df)
-        k =+1
-    # Commit featuresset
-    resp = api.commit_features()
-    print("featureset commit response:", resp)
\ No newline at end of file
+    # Commit featuresets
+        resp = api.commit_featureset(name=fs[k], df=df, metadata=featureset_metadata)
```

Comment:
You could simplify this further by offloading metadata computation to sdk. Add a comment to this affect. You can remove the computation code above and mention that SDK derives metadata from df and put a line reference to sdk code.